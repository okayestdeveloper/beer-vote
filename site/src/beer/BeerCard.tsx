import React from 'react';
import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardHeader from '@material-ui/core/CardHeader';
import Chip from '@material-ui/core/Chip';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import Forward from '@material-ui/icons/Forward';
import { green } from '@material-ui/core/colors';

import {
  IBeer,
  IBeerCardProps,
  EBeerStatus,
  EServingFormat,
  BEER_COLOR_MAP,
} from './beerTypes';
import { useProfile, IProfile } from '../hooks/useProfile';

// icons
import bottle12 from './assets/12oz bottle.png';
import bottle22 from './assets/22oz bottle.png';
import bottle500 from './assets/500ml bottle.png';
import bottle750 from './assets/750ml bottle.png';
import can16 from './assets/16oz can.png';
import can12 from './assets/12oz can.png';
import draught from './assets/draught.png';

const servingImageMap: Map<EServingFormat, any> = new Map([
  [EServingFormat.BOTTLE_12_OZ, bottle12],
  [EServingFormat.BOTTLE_22_OZ, bottle22],
  [EServingFormat.BOTTLE_500_ML, bottle500],
  [EServingFormat.BOTTLE_750_ML, bottle750],
  [EServingFormat.CAN_12_OZ, can12],
  [EServingFormat.CAN_16_OZ, can16],
  [EServingFormat.DRAUGHT, draught],
]);

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    beerCard: {
      width: '320px',
      margin: theme.spacing(1),
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'flex-start',
    },
    beerCardHeader: {
      maxHeight: '3.5rem',
    },
    beerCardContent: {
      flexGrow: 1,
    },
    tagArray: {
      margin: theme.spacing(1, 0),
    },
    tagChip: {
      margin: theme.spacing(0.25),
    },
    servingFormats: {
      margin: theme.spacing(1, 0),
    },
    servingFormatImage: {
      maxHeight: '85px',
      margin: theme.spacing(0.25),
    },
    actionArea: {
      display: 'flex',
      justifyContent: 'flex-end',
    },
    iconButton: {
      cursor: 'pointer',
    },
    upvoteIcon: {
      color: green[700],
      transform: 'rotate(270deg)',
    },
  }),
);

function BeerCard({ beer, upvote }: IBeerCardProps) {
  const classes = useStyles();
  const [profile] = useProfile();
  const titleStyle = {
    backgroundColor: BEER_COLOR_MAP[beer.srm - 1],
    color: beer.srm < 11 ? 'black' : 'white',
  };

  function handleUpvote() {
    if (profile?.email) {
      upvote(beer, profile);
    } else {
      // todo: snackbar
      window.alert('You must sign in to vote.');
    }
  }

  function abv(beer: IBeer): string {
    if (beer.status === EBeerStatus.AVAILABLE) {
      return beer.ABV ? `${beer.ABV.toFixed(1)}%` : '';
    }

    return `${beer.minABV.toFixed(1)}% - ${beer.maxABV.toFixed(1)}% ABV`;
  }

  function tags(beer: IBeer): React.ReactElement[] | null {
    if (!beer.tags) {
      return null;
    }

    return beer.tags.map((tag) => (
      <Chip className={classes.tagChip} label={tag} size="small" key={tag} />
    ));
  }

  function servingFormats(beer: IBeer): React.ReactElement[] | null {
    if (!beer.servingFormats?.length || beer.status !== EBeerStatus.AVAILABLE) {
      return null;
    }

    return beer.servingFormats.map((format) => {
      return (
        <img
          src={servingImageMap.get(format)}
          alt={format}
          className={classes.servingFormatImage}
          key={format}
        />
      );
    });
  }

  function upvoteButton(_profile: IProfile | null): React.ReactElement {
    if (_profile?.email) {
      return (
        <IconButton
          aria-label="upvote"
          onClick={handleUpvote}
          data-testid="upvoteButton"
          className={classes.iconButton}
        >
          <Forward className={classes.upvoteIcon} />
        </IconButton>
      );
    }

    return <></>;
  }

  function actions(beer: IBeer, _profile: IProfile | null): React.ReactElement {
    if (beer.status === EBeerStatus.VOTING) {
      return (
        <>
          <Typography variant="body2" color="textPrimary" component="p">
            {beer.votes} Votes
          </Typography>
          {upvoteButton(_profile)}
        </>
      );
    }
    return <></>;
  }

  // todo: add some indicator of a beer that's being brewed or available? Maybe configurable by props...
  // note: any other modifications for beers available vs. brewing or voting?
  // todo: an image from cloud storage.

  return (
    <Card className={classes.beerCard} raised={true} data-testid="beercard">
      <CardHeader
        title={beer.name}
        subheader={beer.style}
        style={titleStyle}
        subheaderTypographyProps={{ color: 'inherit' }}
      ></CardHeader>
      <CardContent className={classes.beerCardContent}>
        <Typography
          variant="body2"
          color="textPrimary"
          component="p"
          gutterBottom={true}
        >
          {beer.description}
        </Typography>
        <Typography
          variant="body2"
          color="textPrimary"
          component="p"
          gutterBottom={true}
        >
          {abv(beer)}
        </Typography>
        <div className={classes.tagArray}>{tags(beer)}</div>
        <div className={classes.servingFormats}>{servingFormats(beer)}</div>
      </CardContent>
      <CardActions className={classes.actionArea}>
        {actions(beer, profile)}
      </CardActions>
    </Card>
  );
}

export default BeerCard;
