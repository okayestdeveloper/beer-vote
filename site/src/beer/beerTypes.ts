import { IProfile } from './../hooks/useProfile';

export const BEER_COLOR_MAP: Array<string> = [
  '#FFE699',
  '#FFD878',
  '#FFCA5A',
  '#FFBF42',
  '#FBB123',
  '#F8A600',
  '#F39C00',
  '#EA8F00',
  '#E58500',
  '#DE7C00',
  '#D77200',
  '#CF6900',
  '#CB6200',
  '#C35900',
  '#BB5100',
  '#B54C00',
  '#B04500',
  '#A63E00',
  '#A13700',
  '#9B3200',
  '#952D00',
  '#8E2900',
  '#882300',
  '#821E00',
  '#7B1A00',
  '#771900',
  '#701400',
  '#6A0E00',
  '#660D00',
  '#5E0B00',
  '#5A0A02',
  '#600903',
  '#520907',
  '#4C0505',
  '#470606',
  '#440607',
  '#3F0708',
  '#3B0607',
  '#3A070B',
  '#36080A',
  '#000000',
  '#000000',
  '#000000',
  '#000000',
  '#000000',
  '#000000',
  '#000000',
  '#000000',
  '#000000',
  '#000000',
];

export enum EBeerStatus {
  VOTING = 'voting',
  DRAFT = 'draft',
  AVAILABLE = 'available',
  BREWING = 'brewing',
}

export enum EServingFormat {
  BOTTLE_12_OZ = '12oz bottle',
  BOTTLE_22_OZ = '22oz bottle',
  BOTTLE_500_ML = '500ml bottle',
  BOTTLE_750_ML = '750ml bottle',
  CAN_12_OZ = '12oz can',
  CAN_16_OZ = '16oz can',
  DRAUGHT = 'draught',
}

export interface IBeer {
  id: string;
  ABV?: number;
  maxABV: number;
  minABV: number;
  description?: string;
  name: string;
  style: string;
  servingFormats?: EServingFormat[];
  tags?: string[];
  status: EBeerStatus;
  votes: number;
  srm: number;
}

// Props
export interface IBeerListProps {
  beers: IBeer[];
  upvoteBeer: (beer: IBeer, profile: IProfile) => void;
}

export interface IBeerCardProps {
  beer: IBeer;
  upvote: (beer: IBeer, profile: IProfile) => void;
}
