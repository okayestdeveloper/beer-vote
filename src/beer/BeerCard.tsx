import React from 'react';
import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardHeader from '@material-ui/core/CardHeader';
import Chip from '@material-ui/core/Chip';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import Forward from '@material-ui/icons/Forward';
import { green } from '@material-ui/core/colors';

import {
  IBeer,
  IBeerCardProps,
  EBeerStatus,
  EServingFormat,
  BEER_COLOR_MAP,
} from './beerTypes';
import { useProfile } from '../hooks/useProfile';
import bottle12 from './assets/12oz bottle.png';
import bottle22 from './assets/22oz bottle.png';
import bottle500 from './assets/500ml bottle.png';
import bottle750 from './assets/750ml bottle.png';
import can16 from './assets/16oz can.png';
import can12 from './assets/12oz can.png';
import draught from './assets/draught.png';

const servingImageMap: Map<EServingFormat, any> = new Map([
  [EServingFormat.BOTTLE_12_OZ, bottle12],
  [EServingFormat.BOTTLE_22_OZ, bottle22],
  [EServingFormat.BOTTLE_500_ML, bottle500],
  [EServingFormat.BOTTLE_750_ML, bottle750],
  [EServingFormat.CAN_12_OZ, can12],
  [EServingFormat.CAN_16_OZ, can16],
  [EServingFormat.DRAUGHT, draught],
]);

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    beerCard: {
      width: '320px',
      margin: theme.spacing(1),
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'flex-start',
    },
    beerCardHeader: {
      maxHeight: '3.5rem',
    },
    beerCardContent: {
      flexGrow: 1,
    },
    tagArray: {
      margin: theme.spacing(1, 0),
    },
    tagChip: {
      margin: theme.spacing(0.25),
    },
    servingFormats: {
      margin: theme.spacing(1, 0),
    },
    servingFormatImage: {
      maxHeight: '85px',
      margin: theme.spacing(0.25),
    },
    actionArea: {
      display: 'flex',
      justifyContent: 'flex-end',
    },
    iconButton: {
      cursor: 'pointer',
    },
    upvoteIcon: {
      color: green[700],
      transform: 'rotate(270deg)',
    },
  }),
);

const BeerCard: React.FC<IBeerCardProps> = ({ beer, upvote }) => {
  const classes = useStyles();
  const [profile] = useProfile();
  const titleStyle = {
    backgroundColor: BEER_COLOR_MAP[beer.srm - 1],
    color: beer.srm < 11 ? 'black' : 'white',
  };

  const handleUpvote = () => {
    upvote(beer);
  };

  const abv = (beer: IBeer): string => {
    if (beer.status === EBeerStatus.AVAILABLE) {
      return beer.ABV ? `${beer.ABV.toFixed(1)}%` : '';
    }

    return `${beer.minABV.toFixed(1)}% - ${beer.maxABV.toFixed(1)}% ABV`;
  };

  const tags = (beer: IBeer): React.ReactElement[] | null => {
    if (!beer.tags) {
      return null;
    }

    return beer.tags.map((tag) => (
      <Chip className={classes.tagChip} label={tag} size="small" key={tag} />
    ));
  };

  const servingFormats = (beer: IBeer): React.ReactElement[] | null => {
    if (!beer.servingFormats?.length || beer.status !== EBeerStatus.AVAILABLE) {
      return null;
    }

    return beer.servingFormats.map((format) => {
      return (
        <img
          src={servingImageMap.get(format)}
          alt={format}
          className={classes.servingFormatImage}
          key={format}
        />
      );
    });
  };

  const actions = (beer: IBeer): React.ReactElement => {
    if (beer.status === EBeerStatus.VOTING) {
      return (
        <>
          <Typography variant="body2" color="textPrimary" component="p">
            {beer.votes} Votes
          </Typography>
          <IconButton
            aria-label="upvote"
            onClick={handleUpvote}
            data-testid="upvoteButton"
            disabled={profile?.email ? true : false}
            className={classes.iconButton}
          >
            <Forward className={classes.upvoteIcon} />
          </IconButton>
        </>
      );
    }
    return <></>;
  };

  // todo: add some indicator of a beer that's being brewed or available? Maybe configurable by props...
  // note: any other modifications for beers available vs. brewing or voting?
  // todo: tags (pills), serving formats (images/icons). Avatar? Colored header vaguely based on SRM?
  // todo: an image from cloud storage.

  return (
    <Card className={classes.beerCard} raised={true} data-testid="beercard">
      <CardHeader
        title={beer.name}
        subheader={beer.style}
        style={titleStyle}
        subheaderTypographyProps={{ color: 'inherit' }}
      ></CardHeader>
      <CardContent className={classes.beerCardContent}>
        <Typography
          variant="body2"
          color="textPrimary"
          component="p"
          gutterBottom={true}
        >
          {beer.description}
        </Typography>
        <Typography
          variant="body2"
          color="textPrimary"
          component="p"
          gutterBottom={true}
        >
          {abv(beer)}
        </Typography>
        <div className={classes.tagArray}>{tags(beer)}</div>
        <div className={classes.servingFormats}>{servingFormats(beer)}</div>
      </CardContent>
      <CardActions className={classes.actionArea}>{actions(beer)}</CardActions>
    </Card>
  );
};

export default BeerCard;
