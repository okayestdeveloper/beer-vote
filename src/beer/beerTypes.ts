export enum EBeerStatus {
  VOTING = 'voting',
  DRAFT = 'draft',
  AVAILABLE = 'available',
  BREWING = 'brewing',
}

export enum EServingFormat {
  BOTTLE_12_OZ = '12oz bottle',
  BOTTLE_22_OZ = '22oz bottle',
  BOTTLE_500_ML = '500ml bottle',
  BOTTLE_750_ML = '750ml bottle',
  CAN_12_OZ = '12oz can',
  CAN_16_OZ = '16oz can',
  DRAUGHT = 'draught',
}

export interface IBeer {
  id: string;
  ABV?: number;
  maxABV: number;
  minABV: number;
  description?: string;
  name: string;
  style: string;
  servingFormats?: EServingFormat[];
  tags?: string[];
  status: EBeerStatus;
  votes: number;
}

// Props
export interface IBeerListProps {
  beers: IBeer[];
  upvoteBeer: (beer: IBeer) => void;
}

export interface IBeerCardProps {
  beer: IBeer;
  upvote: (beer: IBeer) => void;
}
