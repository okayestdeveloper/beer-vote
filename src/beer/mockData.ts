import { IBeer, EServingFormat, EBeerStatus } from './beerTypes';

const beers: IBeer[] = [
  {
    id: 'beer1',
    maxABV: 7.0,
    minABV: 6.0,
    description: 'A dark chocolate milk stout.',
    name: 'Valkyrie',
    style: 'Chocolate Stout',
    servingFormats: [EServingFormat.BOTTLE_12_OZ, EServingFormat.CAN_12_OZ],
    tags: ['dark', 'chocolate', 'rich'],
    status: EBeerStatus.VOTING,
    votes: 543,
  },
  {
    id: 'beer2',
    ABV: 5.5,
    maxABV: 4.0,
    minABV: 6.0,
    description: 'An American Pale Ale with tropical, fruity hops.',
    name: 'Seeker',
    style: 'Juicy Pale Ale',
    servingFormats: [EServingFormat.DRAUGHT],
    tags: ['juicy', 'bright', 'refreshing'],
    status: EBeerStatus.AVAILABLE,
    votes: 32,
  },
];

export const getBeers = (): IBeer[] => {
  return JSON.parse(JSON.stringify(beers));
};
