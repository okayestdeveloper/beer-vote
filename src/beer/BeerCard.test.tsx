import React from 'react';
import {
  cleanup,
  render,
  fireEvent,
  RenderResult,
} from '@testing-library/react';
import renderer from 'react-test-renderer';

import { IBeer, IBeerCardProps } from './beerTypes';
import BeerCard from './BeerCard';
import { getBeers } from './mockData';

describe('BeerCard rendering', () => {
  let beers: IBeer[];

  beforeEach(() => {
    beers = getBeers();
  });

  it(`should render a card for a beer that's in voting status`, () => {
    const tree = renderer.create(
      <BeerCard beer={beers[0]} upvote={jest.fn()} />,
    );
    expect(tree).toMatchSnapshot();
  });

  it(`should render a card for a beer that's in available status`, () => {
    const tree = renderer.create(
      <BeerCard beer={beers[1]} upvote={jest.fn()} />,
    );
    expect(tree).toMatchSnapshot();
  });
});

describe('BeerCard props', () => {
  let card: RenderResult;
  let mockUpvote: jest.Mock;
  let beers: IBeer[];

  function renderCard(args?: Partial<IBeerCardProps>): RenderResult {
    const defaultProps: IBeerCardProps = {
      beer: beers[0],
      upvote: () => {},
    };

    const props = { ...defaultProps, ...args };
    return render(<BeerCard {...props} />);
  }

  beforeEach(() => {
    beers = getBeers();
    mockUpvote = jest.fn();
    card = renderCard({ upvote: mockUpvote });
  });

  afterEach(cleanup);

  it(`should call the upvote when user clicks the button`, () => {
    const btn = card.getByTestId('upvoteButton');
    fireEvent.click(btn);
    expect(mockUpvote).toHaveBeenCalled();
  });
});
