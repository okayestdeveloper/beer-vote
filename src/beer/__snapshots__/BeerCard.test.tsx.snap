// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerCard rendering should render a card for a beer that's in available status 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-beerCard-1 MuiPaper-elevation8 MuiPaper-rounded"
  data-testid="beercard"
>
  <div
    className="MuiCardHeader-root"
    style={
      Object {
        "backgroundColor": "#DE7C00",
        "color": "black",
      }
    }
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        Seeker
      </span>
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorInherit MuiTypography-displayBlock"
      >
        Juicy Pale Ale
      </span>
    </div>
  </div>
  <div
    className="MuiCardContent-root makeStyles-beerCardContent-3"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-gutterBottom"
    >
      An American Pale Ale with tropical, fruity hops.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-gutterBottom"
    >
      5.5%
    </p>
    <div
      className="makeStyles-tagArray-4"
    >
      <div
        className="MuiChip-root makeStyles-tagChip-5 MuiChip-sizeSmall"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label MuiChip-labelSmall"
        >
          juicy
        </span>
      </div>
      <div
        className="MuiChip-root makeStyles-tagChip-5 MuiChip-sizeSmall"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label MuiChip-labelSmall"
        >
          bright
        </span>
      </div>
      <div
        className="MuiChip-root makeStyles-tagChip-5 MuiChip-sizeSmall"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label MuiChip-labelSmall"
        >
          refreshing
        </span>
      </div>
    </div>
    <div
      className="makeStyles-servingFormats-6"
    >
      <img
        alt="draught"
        className="makeStyles-servingFormatImage-7"
        src="draught.png"
      />
    </div>
  </div>
  <div
    className="MuiCardActions-root makeStyles-actionArea-8 MuiCardActions-spacing"
  />
</div>
`;

exports[`BeerCard rendering should render a card for a beer that's in voting status 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-beerCard-1 MuiPaper-elevation8 MuiPaper-rounded"
  data-testid="beercard"
>
  <div
    className="MuiCardHeader-root"
    style={
      Object {
        "backgroundColor": "#000000",
        "color": "white",
      }
    }
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        Valkyrie
      </span>
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorInherit MuiTypography-displayBlock"
      >
        Chocolate Stout
      </span>
    </div>
  </div>
  <div
    className="MuiCardContent-root makeStyles-beerCardContent-3"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-gutterBottom"
    >
      A dark chocolate milk stout.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-gutterBottom"
    >
      6.0% - 7.0% ABV
    </p>
    <div
      className="makeStyles-tagArray-4"
    >
      <div
        className="MuiChip-root makeStyles-tagChip-5 MuiChip-sizeSmall"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label MuiChip-labelSmall"
        >
          dark
        </span>
      </div>
      <div
        className="MuiChip-root makeStyles-tagChip-5 MuiChip-sizeSmall"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label MuiChip-labelSmall"
        >
          chocolate
        </span>
      </div>
      <div
        className="MuiChip-root makeStyles-tagChip-5 MuiChip-sizeSmall"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
      >
        <span
          className="MuiChip-label MuiChip-labelSmall"
        >
          rich
        </span>
      </div>
    </div>
    <div
      className="makeStyles-servingFormats-6"
    />
  </div>
  <div
    className="MuiCardActions-root makeStyles-actionArea-8 MuiCardActions-spacing"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
    >
      543
       Votes
    </p>
    <button
      aria-label="upvote"
      className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-9"
      data-testid="upvoteButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root makeStyles-upvoteIcon-10"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8V4l8 8-8 8v-4H4V8z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
